powergenome/load_construction.py

```diff
@@ -0,0 +1,673 @@
+states_central = list(map(abbr2state, states_central_abbr))
+states_mountain = list(map(abbr2state, states_mountain_abbr))
+states_pacific = list(map(abbr2state, states_pacific_abbr))
+
+# some parameters
+stated_states = ["New Jersey", "New York", "Virginia"]
+# Date Jan 29, 2021
+# (2) PA, NJ, VA, NY, MI all have EV and heat pump stocks from NZA DD case
+# consistent with their economywide decarbonization goals.
+# https://www.c2es.org/content/state-climate-policy/
+# Date Feb 10, 2021
+# Remove high electrification growth in PA and MI in stated policies;
+# they dont have clean energy goals so kind of confusing/inconsistent to require high electrification in these states.
+# So our new "Stated Policies" definition for electrification is states
+# with BOTH economywide emissions goals + 100% carbon-free electricity standards
+# = NY, NJ, VA.
+stated_states_abbr = list(map(state2abbr, stated_states))
+# years = ["2022", "2025", "2030", "2040", "2050"]
+cases = ["current_policy", "stated_policy", "deep_decarbonization"]
+
+running_sector = [
+    "Residential",
+    "Residential",
+    "Commercial",
+    "Commercial",
+    "Transportation",
+    "Transportation",
+    "Transportation",
+    "Transportation",
+]
+running_subsector = [
+    "space heating and cooling",
+    "water heating",
+    "space heating and cooling",
+    "water heating",
+    "light-duty vehicles",
+    "medium-duty trucks",
+    "heavy-duty trucks",
+    "transit buses",
+]
+Nsubsector = len(running_subsector)
+
+logger = logging.getLogger(__name__)
+
+
+def CreateOutputFolder(case_folder):
+    path = case_folder / "extra_outputs"
+    if not os.path.exists(path):
+        os.makedirs(path)
+
+
+######################################
+# CreatingBaseLoad.R
+def CreateBaseLoad(
+    years: List[int],
+    regions: List[str],
+    output_folder: Path,
+    path_growthrate: Path = None,
+) -> None:
+    path_processed = path_in
+    path_result = output_folder
+
+    ## Method 3: annually
+    load_dtypes = {
+        "Year": "category",
+        "LocalHourID": "category",
+        "Sector": "category",
+        "Subsector": "category",
+        "LoadMW": np.float32,
+    }
+    model_states = pop.loc[pop["ipm_region"].isin(regions), "state"]
+    EFS_2020_LoadProf = pd.read_parquet(path_in / "EFS_REF_load_2020.parquet")
+    total_efs_2020_load = EFS_2020_LoadProf["LoadMW"].sum()
+
+    EFS_2020_LoadProf = EFS_2020_LoadProf.loc[
+        EFS_2020_LoadProf["State"].isin(model_states), :
+    ]
+    EFS_2020_LoadProf = EFS_2020_LoadProf.astype(load_dtypes)
+    EFS_2020_LoadProf = pd.merge(EFS_2020_LoadProf, pop, on=["State"])
+    EFS_2020_LoadProf = EFS_2020_LoadProf.assign(
+        weighted=EFS_2020_LoadProf["LoadMW"] * EFS_2020_LoadProf["State Prop"]
+    )
+    EFS_2020_LoadProf = EFS_2020_LoadProf.groupby(
+        ["Year", "ipm_region", "LocalHourID", "Sector", "Subsector"],
+        as_index=False,
+        observed=True,
+    ).agg({"weighted": "sum"})
+
```

Comment:
@ritibhandarkar and @xuqingyu why are we using the calculated 2019 demand to calculate base load? Is that the basis year for EFS data?